File: desktop/python/binary_decimal.py
#********************decimal to binary*********************
'''n=int(input("enter the decimal num"))
binary=0
base=1
while(n>0):
    rem=n%2
    binary=binary+rem*base
    n=n//2
    base=base*10
print(binary)'''
#*********************binary to decimal**********************
'''n=int(input("enter the binary num"))
deci=0
base=1
while(n>0):
    rem=n%10
    deci=deci+rem*base
    n=n//10
    base=base*2
print(deci)'''
#****************another method**********************
n=int(input("enter the num"))
bin=""
while(n>0):
    r=n%2
    n=n//2
    bin=str(r)+bin
print(bin)

File: desktop/dikshith/binary_decimal.c
//  binary to decimal
/*#include<stdio.h>
void main()
{
	int n,rem,binary,dec=0,base=1;
	printf("enter the binary num\n");
	scanf("%d",&binary);
	while(binary>0)
	{
		rem=binary%10;
		dec=dec+rem*base;
		binary=binary/10;
		base=base*2;
	}
	printf("%d",dec);
}*/
// decimal to binary
/*#include<stdio.h>
void main()
{
        int n,rem,binary,dec=0,base=1;
        printf("enter the binary num\n");
        scanf("%d",&binary);
        while(binary>0)
        {
                rem=binary%2;
                dec=dec+rem*base;
                binary=binary/2;
                base=base*10;
        }
        printf("%d",dec);
}*/
// decimal to octal
#include<stdio.h>
/*void main()
{
        int n,rem,binary,dec=0,base=1;
        printf("enter the binary num\n");
        scanf("%d",&binary);
        while(binary>0)
        {
                rem=binary%8;
                dec=dec+rem*base;
                binary=binary/8;
                base=base*10;
        }
        printf("%d",dec);
}*/
//binary to octal
#include<stdio.h>
void main()
{
        int n,rem,binary,dec=0,base=1;
        printf("enter the binary num\n");
        scanf("%d",&binary);
        while(binary>0)
        {
                rem=binary%2;
                dec=dec+rem*base;
                binary=binary/2;
                base=base*10;
        }
        printf("%d",dec);

}
